src/gui/base/Dialog.ts

```diff
@@ -1006,6 +1007,30 @@ export class Dialog implements ModalComponent {
 		return dialog
 	}
 
+	/**
+	 * Show a dialog with a preview of a given list of contacts
+	 * @param contacts The contact list to be previewed
+	 * @param okAction The action to be executed when the user press the import button
+	 */
+	static importVCardDialog(contacts: Contact[], okAction: (dialog: Dialog) => unknown): Dialog {
+		return Dialog.showActionDialog({
+			title: () => lang.get("importVCard_action"),
+			child: () =>
+				m(
+					".mt-s",
+					contacts.map((contact) => {
```

Comment:
Since we are going to use it from native import as well I was wondering if it would be a better idea to use a real `List` instead. We just need to give it a static state, we don't need a `ListModel`

Reply:
Implemented